packages/testing/src/protos/google/firebase/rules/v1/source.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package google.firebase.rules.v1;
+
+option java_package = "com.google.firebase.rules.v1";
+option java_multiple_files = true;
+option java_outer_classname = "SourceProto";
+
+// `Source` is one or more `File` messages comprising a logical set of rules.
+message Source {
+  // `File` set constituting the `Source` bundle.
+  repeated File files = 1;
+
+  // `Language` of the `Source` bundle. If unspecified, the language will
+  // default to `FIREBASE_RULES`.
+  Language language = 2;
+}
+
+// `File` containing source content.
+message File {
+  // Textual Content.
+  string content = 1;
+
+  // File name.
+  string name = 2;
+
+  // Fingerprint (e.g. github sha) associated with the `File`.
+  bytes fingerprint = 3;
+}
+
+// Position in the `Source` content including its line, column number, and an
+// index of the `File` in the `Source` message. Used for debug purposes.
+message SourcePosition {
```

Comment:
this doesn't look to be used

Reply:
This is a pre-existing proto file. I don't want to modify it.

Reply:
i see, where does this come from? do we need licenses or anything?

Reply:
http://google3/google/firebase/rules/v1/source.proto

Added the licenses at the top of each proto file, good call.